### PR TITLE
[2.13.x] DDF-3933 Add external links dropdown to Intrigue

### PR DESCRIPTION
**Pull request by JGatley**
#### What does this PR do?
This PR adds a dropdown menu displaying all `associations.external` URL's for metacards in Intrigue that have any `associations.external` attributes. 
#### Who is reviewing it? 
@clockard 
@kcover 
@emmberk 
@peterhuffer 
@oconnormi 

#### Select relevant component teams: 
@codice/ui 
#### Ask 2 committers to review/merge the PR and tag them here.

@andrewkfiedler
@clockard

#### How should this be tested?

- Ingest a metacard.
- In the inspector's details tab, add an associations.external attribute. 
- Edit the associations.external attribute, add URL(s) with a valid protocol.
- In the Intrigue search, verify that the new external links button appears and is usable. Verify that other buttons attached to the metacard were not negatively impacted. 

#### What are the relevant tickets?
[DDF-3933](https://codice.atlassian.net/browse/DDF-3933)

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by JGatley** [comment]
build now

**Comment by JGatley** [comment]
build now

File: ui/packages/catalog-ui-search/src/main/webapp/component/result-link/result-link.hbs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ * version 3 of the License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+ --}}
+<div class="is-header">
+    Follow External Link
```
Comment (clockard) [inline ui/packages/catalog-ui-search/src/main/webapp/component/result-link/result-link.hbs]: "Follow External Link(s)" ?

Comment (JGatley) [reply ui/packages/catalog-ui-search/src/main/webapp/component/result-link/result-link.hbs]: Edited

**Comment by JGatley** [comment]
build now

**Comment by JGatley** [comment]
image

image

**Comment by JGatley** [comment]
Attached screenshots of the button and dropdown generated. Note that long URLs will be given ellipses on overflow to keep the list clean. On mouseover, the full URL is displayed in the tooltip.

File: ui/packages/catalog-ui-search/src/main/webapp/component/visualization/table/row.less [hunk 2]
```diff
@@ -48,6 +48,14 @@
   }
 }
 
+@{customElementNamespace}result-row:not(.is-link) {
```
Comment (andrewkfiedler) [inline ui/packages/catalog-ui-search/src/main/webapp/component/visualization/table/row.less]: ❓ Why are we adding this?

File: ui/packages/catalog-ui-search/src/main/webapp/component/result-link/result-link.less [hunk 3]
```diff
@@ -0,0 +1,14 @@
+@{customElementNamespace}result-link {
+    .customElement;
+    padding: @minimumSpacing;
+    max-width: 800px;
```
Comment (andrewkfiedler) [inline ui/packages/catalog-ui-search/src/main/webapp/component/result-link/result-link.less]: We can probably get by with only putting a max-width on the links below.

File: ui/packages/catalog-ui-search/src/main/webapp/component/result-link/result-link.less [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+    .is-link {
+        line-height: @minimumButtonSize;
+        display: block;
+        max-width: 600px;
```
Comment (andrewkfiedler) [inline ui/packages/catalog-ui-search/src/main/webapp/component/result-link/result-link.less]: I would consider changing this max width to be @minimumScreenSize.

**Comment by JGatley** [comment]
Addressed comments made by @andrewkfiedler

**Comment by JGatley** [comment]
build now

**Comment by kcover** [comment]
Manually tested by uploading a test file and adding 3 external associations with working links. I verified that the dropdown was available, that long links were truncated with ellipses, and that each of the links worked as expected. LGTM 👍

**Comment by JGatley** [comment]
build now

**Comment by clockard** [comment]
there were unrelated itests failures 
build now

**Pull request merged**